networkit/cpp/graph/TopologicalSort.cpp

```diff
@@ -4,40 +4,94 @@
  *  Created on: 22.11.2021
  *      Author: Fabian Brandt-Tumescheit
  */
+#include <stdexcept>
+#include "networkit/graph/GraphTools.hpp"
 #include <networkit/graph/TopologicalSort.hpp>
 
 namespace NetworKit {
 
-TopologicalSort::TopologicalSort(const Graph &G) : G(&G) {
-    if (!G.isDirected())
+TopologicalSort::TopologicalSort(const Graph &G)
+    : G(&G), computedNodeIdMap(GraphTools::getContinuousNodeIds(G)) {
```

Comment:
`computedNodeIdMap` get computed twice - once in the intializer list of the constructor and once in line below in the if-case. As discussed, we should avoid the default computation of the list.